code/game/objects/items/RSF.dm

```diff
@@ -8,178 +8,200 @@ RSF
 	desc = "A device used to rapidly deploy service items."
 	icon = 'icons/obj/tools.dmi'
 	icon_state = "rsf"
+	///The icon state to revert to when the tool is empty
+	var/spent_icon_state = "rsf_empty"
 	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
 	opacity = 0
 	density = FALSE
 	anchored = FALSE
 	item_flags = NOBLUDGEON
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 0)
+	///The current matter count
 	var/matter = 0
-	var/mode = 1
+	///The max amount of matter in the device
+	var/max_matter = 30
+	///The type of the object we are going to despense
+	var/to_despense
```

Comment:
why `d**E**spense`

Reply:
typo
`var/to_dispense`
fix this for all other instances of this var

Reply:
and in dmdoc

Reply:
I'm a dumbass, I'll fix it when I get to the rest of the issues here.